app/code/community/Bolt/Boltpay/OrderCreationException.php

```diff
@@ -0,0 +1,214 @@
+    const E_BOLT_SHIPPING_PRICE_HAS_BEEN_UPDATED_TMPL      = '{"reason": "Shipping total has changed", "old_value": "%d", "new_value": "%d"}';
+
+    /**
+     * @var int http response code that is to be returned
+     */
+    protected $httpCode;
+
+    /**
+     * @var string The json to be returned to Bolt associated with this exception
+     */
+    protected $json;
+
+    /**
+     * @var array All possible error codes supported by Bolt
+     */
+    protected static $validCodes = array(
+        self::E_BOLT_GENERAL_ERROR,
+        self::E_BOLT_ORDER_ALREADY_EXISTS,
+        self::E_BOLT_CART_HAS_EXPIRED,
+        self::E_BOLT_ITEM_PRICE_HAS_BEEN_UPDATED,
+        self::E_BOLT_OUT_OF_INVENTORY,
+        self::E_BOLT_DISCOUNT_CANNOT_APPLY,
+        self::E_BOLT_DISCOUNT_DOES_NOT_EXIST,
+        self::E_BOLT_SHIPPING_PRICE_HAS_BEEN_UPDATED
+    );
+
+    /**
+     * Bolt_Boltpay_InvalidTransitionException constructor.
+     *
+     * @param int    $code          The Bolt defined error code
+     * @param string $dataTemplate  specific Bolt error sub-category
+     * @param array  $dataValues    An array of values to be added to the data template
+     * @param string $message       The exception message to throw.
+     * @param Throwable|null $previous  [optional] The previously throwable used for exception chaining.
+     */
+    public function __construct($code = self::E_BOLT_GENERAL_ERROR, $dataTemplate = self::E_BOLT_GENERAL_ERROR_TMPL_GENERIC, array $dataValues = array(), $message = null, Throwable $previous = null)
+    {
+        // If code is invalid, we will use the generic message
+        if (!in_array($code, self::$validCodes)) {
+            $code = self::E_BOLT_GENERAL_ERROR;
+            $dataTemplate = self::E_BOLT_GENERAL_ERROR_TMPL_GENERIC;
+            $dataValues = array(addcslashes($message, '"\\'));
+        }
+
+        $this->setHttpCode($code, $dataTemplate);
+        $this->setJson($code, $dataTemplate, $dataValues);
+
+        if (empty($message)) {
+            $message = $this->getJson();
+        }
+
+        parent::__construct($message, $code, $previous);
+    }
+
+    /**
+     * Sets the httpCode based on the error
+     *
+     * @param int    $code         Bolt error code
+     * @param string $dataTemplate specific Bolt error sub-category
+     *
+     * @return int The HTTP code that was set
+     */
+    private function setHttpCode( $code, $dataTemplate ) {
+        // Select the http code
+        switch ($code) {
+            case self::E_BOLT_GENERAL_ERROR:
+                if ($dataTemplate === self::E_BOLT_GENERAL_ERROR_TMPL_HMAC) {
+                    $this->httpCode = 401;
+                    break;
+                }
+            case self::E_BOLT_ORDER_ALREADY_EXISTS:
+                $this->httpCode = 409;
+                break;
+            case self::E_BOLT_CART_HAS_EXPIRED_TMPL_NOT_FOUND:
+                $this->httpCode = 404;
```

Comment:
does magento or php have constant for these HTTP status code? If possible, I prefer we say `return StatusCodes.NotFound` instead of `return 404`

Reply:
Maybe.  But, if not, we will implement it.  I agree.  Here it will help the engineer understand the HTTP spec'd significance of each error code.  This is a non-imperative beautification request.

Task tracked here:
https://app.asana.com/0/1118494470563243/1121207201774763